{{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/analytics/{%if cookiecutter.use_analytics == 'y'%}views.py{%endif%}

```diff
@@ -0,0 +1,114 @@
+from django.dispatch import receiver
+from rest_framework import status
+from rest_framework.decorators import (
+    api_view,
+    authentication_classes,
+    permission_classes,
+)
+from rest_framework.response import Response
+
+import analytics
+from django.contrib.auth.signals import user_logged_in
+
+from {{cookiecutter.repo_name}}.analytics.serializers import (
+    AliasSerializer,
+    GroupSerializer,
+    IdentifySerializer,
+    TrackSerializer,
+    ViewEventSerializer,
+)
+from {{cookiecutter.repo_name}}.analytics.utils import get_analytics_payload
+
+
+# Alias call for Segment
+def alias(request):
+    serializer = AliasSerializer(data=request.data, context={"request": request})
+
+    if serializer.is_valid():
+        payload = get_analytics_payload(serializer.data, "alias")
+        analytics.alias(*payload)
+        return True
+
+    return False
+
+
+# Identify call for Segment
+def identify(request):
+    serializer = IdentifySerializer(data=request.data, context={"request": request})
+
+    if serializer.is_valid():
+        payload = get_analytics_payload(serializer.data, "identify")
+        analytics.identify(*payload)
+        return True
+
+    return False
+
+
+# Identify call for Segment
+@receiver(user_logged_in)
+def login_or_register(request, **kwargs):
+    return alias(request) and identify(request)
+
+
+# Group call for Segment
+@api_view(["POST"])
+@authentication_classes([])
+@permission_classes([])
+def group(request):
+    serializer = GroupSerializer(data=request.data, context={"request": request})
```

Comment:
This is a better use for the serializer since it's directly touching untrusted data from the outside.